clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/CloudFoundryClientUtilsTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 Pivotal, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.cloudfoundry.client;
+
+import com.netflix.spinnaker.clouddriver.cloudfoundry.client.api.ApplicationService;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.client.model.v3.Application;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.client.model.v3.Pagination;
+import org.junit.jupiter.api.Test;
+
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+class CloudFoundryClientUtilsTest {
+
+  @Test
+  void collectPagesIteratesOverMultiplePages(){
+    ApplicationService applicationService = mock(ApplicationService.class);
+    Application applicationOne =  new Application().setName("app-name-one").setGuid("app-guid-one").setState("STARTED").setCreatedAt(ZonedDateTime.now()).setLinks(Collections.emptyMap());
+    List pageOneResources = Collections.singletonList(applicationOne);
+    Pagination<Application> pageOne = new Pagination<>();
+    pageOne.setPagination(new Pagination.Details().setTotalPages(2));
+    pageOne.setResources(pageOneResources);
+    Application applicationTwo =  new Application().setName("app-name-two").setGuid("app-guid-two").setState("STARTED").setCreatedAt(ZonedDateTime.now()).setLinks(Collections.emptyMap());
+    List pageTwoResources = Collections.singletonList( applicationTwo );
+    Pagination<Application> pageTwo = new Pagination<>();
+    pageTwo.setPagination(new Pagination.Details().setTotalPages(2));
+    pageTwo.setResources(pageTwoResources);
+
+    when(applicationService.all(null, null,null)).thenReturn(pageOne);
+    when(applicationService.all(2, null,null)).thenReturn(pageTwo);
+
+    List results = CloudFoundryClientUtils.collectPages("applications", page -> applicationService.all(page, null, null));
+
+    assertThat(results.size()).isEqualTo(2);
+    assertThat(results).contains(applicationOne, applicationTwo);
```

Comment:
you should be able to do just one assert for equality on the list

Reply:
specifically, call `containsExactly` instead of `contains` and remove the size assertion.